src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTokenExtensions.cs

```diff
@@ -16,6 +16,9 @@ namespace Microsoft.CodeAnalysis.CSharp.Extensions
 {
     internal static partial class SyntaxTokenExtensions
     {
+        public static void Deconstruct(this SyntaxToken token, out SyntaxKind kind)
```

Comment:
Is this used anywhere?

Reply:
yes.  it's used in teh pattern matching: `if (argument is { RefOrOutKeyword: SyntaxToken(SyntaxKind.OutKeyword) })`